### DISCUSSION TITLE
Moving Laragon along...

### DISCUSSION THREAD

**Discussion by Sophist-UK** [leokhoa/laragon#889] [General]
As someone who has been around in various open source projects for about a decade and a half, I do understand the difficulties of a lone developer trying to maintain **and** enhance their open source project whilst holding down a day job and attempting to have a life outside work and open source.

However this can result in a great open source project which isn't responsive to user needs and only gets updated once every few years in a spurt of activity.

I would therefore like to propose to Leo @leokhoa that he might like to think about how he can broaden the scope for users to contribute to the development of Laragon (and thus harness the mass of time and effort this makes available to the project) whilst at the same time (and most importantly) allowing him to retain control over both the strategic direction and the quality of Laragon within the small amount of time he **can** commit on a day-to-day basis.

In my experience with open source, there is often a willing community of volunteers who, in return for getting to use a great tool, are are very willing to give something back in order to make the tool even greater. And in my view, there are a number of ways this can be achieved:

1. Publicly setting a strategic direction, creating a To-Do list, encouraging users to submit PRs to deliver items on the To-Do list and providing prompt reviews and decisions on these PRs.

2. Trying to find ways to delegate the full lifecycle of **documentation** enhancement to others in order to focus the core time available to code reviews etc.

3. Based on the contributions made in the early days, find community members who can start to review the PRs of others and share the review workload. As these more senior members demonstrate their abilities, perhaps e.g. allow them to approve PRs rather than just review them etc.

I hope that Leo is willing to consider these suggestions in the positive way that they are intended.

**Comment by AWqxKAWERbXo** [comment]
As I offered earlier, I'd support managing the issues section to reduce the number of ~250 to an amount that can be handled and is up to date with the current needs. I already did that for years for <https://github.com/FriendsOfREDAXO/>.

My two cents: Trustworthyness of contributors should be `#1`, as we saw that on XZ. <https://www.youtube.com/watch?v=F7iLfuci75Y> <https://en.wikipedia.org/wiki/XZ_Utils_backdoor>. That said, I'm sure many people inside the community are trustworthy and there are lots of tasks to do. Tasks, that are non-criticial for security. To make things clear: I don't want to install a version of laragon that comes with a credentials scanner or 🔐 ransomware.

**Reply by Sophist-UK** [reply]
I couldn't agree more - but this is why PRs are mandatory and proper reviews & merging need to be done by a limited trusted core team.

But I genuinely believe that if Leo maintained a ToDo list of small individual improvements, the community would deliver them and Leo would only need to review and merge them and could focus the rest of his time on core enabling features in the Laragon controller.

**Comment by Sophist-UK** [comment]
@leokhoa As a starter I would like to nominate @alxndr-w to have some delegated rights on the Github repo. I have had a look at his profile both here on Github and more widely and my own evaluation is that he has an appropriate track record on this project, on other Github projects and more widely in IT (though of course my personal recommendation is only as trustworthy as I am - so research this yourself).

**Comment by Sophist-UK** [comment]
@leokhoa 

I hadn't realised it before, but the source code for the Laragon executable is not shared here i.e. is not open source (even though the executable is uploaded here and is open source in the sense it is free to use).

Before making any other comments, let me say that I do understand why you might have reservations about sharing the source code for the executable - by keeping this private you are keeping control over the project and preventing people creating competing forks of the project.

However, this does create some concerns for me as a user, and I suspect that these are shared by other users too:

1. If you disappear (and there have been recent worries about this) the source code will effectively be lost, and Laragon will wither on the vine as a consequence.
2. If you are finding it difficult to find the time to provide support and keep Laragon current, no one can help you with changes to the executable.

So as part of the suggestions I am asking you to possibly consider, can I add these:

1. Consider making the Laragon executable have less functionality by it providing more generalised functionality that can be flexibly adjusted using configuration files and scripts that are source code available.
2. Consider sharing the source code privately with a limited set of trusted community members who can assist you with coding.
3. Or possibly consider posting the source code openly on Github so that the community can contribute to it.

Thanks.

S
